File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {

    int getLength(ListNode temp){
        ListNode temp1 = temp;
        int count=0;
        while(temp1!=null){
            temp1=temp1.next;
            count++;
        }
        return count;
    }
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
       if(headA == null || headB == null) return null;
       
    //    checking the lenght of the List
    int lengthA= getLength(headA);
    int lengthB = getLength(headB);

    // align both the list to the same level
    while(lengthA > lengthB){
     headA=headA.next;
     lengthA--;
    }

    while(lengthA < lengthB){
     headB=headB.next;
     lengthB--;
    }

    while(headA != null){
        if(headA == headB){
            return headA;
        }
        headA=headA.next;
        headB=headB.next;

    }

    return null;




    }
}

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    

    let node1 = headA;
    let node2 = headB;

    let newSet = new Set();

    while(node2 != null){
     newSet.add(node2);

     node2 = node2.next;
    }



    while(node1 != null){
        if(newSet.has(node1)) return node1;

        node1 = node1.next;
    }


    return null
};